packages/tangle/messagebooker.go

```diff
@@ -183,7 +185,12 @@ func (m *MessageBooker) referencedTransactionIDs(transaction *ledgerstate.Transa
 	return
 }
 
+// Attachments retrieves the attachments of a transaction.
 func (m *MessageBooker) Attachments(transactionID ledgerstate.TransactionID) (attachments MessageIDs) {
+	cachedAttachments := m.messageStore.Attachments(transactionID)
+	cachedAttachments.Consume(func(attachment *Attachment) {
+		attachments = append(attachments, attachment.MessageID())
+	})
 	return
 }
```

Comment:
I think this could be part of `MessageStore`. It could be a convenience function `AttachmentsMessageIDs` or something :)

Reply:
uhmm, I prefer `messageStore` to be lean. But it doesn't matter if we want this convenience function tbh.